app/src/main/java/io/gnosis/safe/utils/ImageUtils.kt

```diff
@@ -0,0 +1,31 @@
+package io.gnosis.safe.utils
+
+import android.widget.ImageView
+import androidx.core.view.setPadding
+import com.squareup.picasso.Picasso
+import io.gnosis.safe.R
+
+fun ImageView.loadTokenLogo(icon: String?) {
+    setPadding(0)
+    background = null
+    setImageDrawable(null)
+    colorFilter = null
+    when {
+        icon == "local::ethereum" -> {
+            setImageResource(R.drawable.ic_ethereum_logo)
+        }
+        icon?.startsWith("local::") == true -> {
+//            setImageResource(R.drawable.circle_background)
+        }
+        !icon.isNullOrBlank() ->
+            Picasso.get()
+                .load(icon)
+//                .placeholder(R.drawable.circle_background)
```

Comment:
why do we have commented out code here?

Reply:
There was a bunch of stuff I didn't paste from the actual method Utils that eventually we might need. I figured I would left it comment as a reminder that the functionality was copied partially. If you want I can carry over the assets and uncomment all this code. I suppose the token icon should behave the same as in old apps, right?

Reply:
yes. It's either icon or some placeholder